### PR TITLE
feat(Select): Add ability to make select full width and have label

### PR DESCRIPTION
**Pull request by ayeshakmaz**
Summary:

Added isFullWidth to make Select take 100% width of container

Added label, isOptional, and isRequired to specify a label for the select ( a big A11Y improvement)

Added onChange, displayError, and note to get parity w/ Input:

Screen Shot 2020-04-02 at 4 43 00 PM
Screen Shot 2020-04-02 at 4 43 07 PM
Screen Shot 2020-04-02 at 4 43 19 PM

### CODE REVIEW

File: src/components/Select/index.js [hunk 1]
```diff
@@ -11,29 +13,61 @@ const Select = ({
...
+        disabled={ isDisabled }
+        required={ isRequired }
+        name={ name }
+        id={ id }
+        className={ classNames('oui-select', {'width--1-1': isFullWidth}) }>
+        {isRequired && <option value="">Please select an option</option>}
+        { children }
+      </select>
+    );
+  };
```
Comment (circAssimilate) [inline src/components/Select/index.js:36]: nit: I don't think this has to be a function: e.g.

```
const renderSelect = (
    <select >
       ...
     </select>
);
```

File: src/components/Select/Select.story.js [hunk 2]
```diff
@@ -4,21 +4,69 @@ import { storiesOf } from '@storybook/react';
...
+  .add('Default', () => {
+    return (
+      <Select
+        label="My favorite zoo animal"
+        isDisabled={ boolean('isDisabled', false) }
+        isOptional={ boolean('isOptional', false) }
+        name="zoo"
+        id="zoo">
+        <option value="one">This is option one - a Lion</option>
+        <option value="two">And this is option two - a Tiger</option>
```
Comment (circAssimilate) [inline src/components/Select/Select.story.js:24]: LOL. I know what someone has been watching on Netflix. 😸

Comment (ayeshakmaz) [reply src/components/Select/Select.story.js:24]: hahha was hoping someone would catch that!

**Comment by ayeshakmaz** [comment]
Sorry @circAssimilate - @acacia-m-pappas and I found more gaps: could you re-review when you get a chance? Thank you!!

**Pull request merged**